integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationTest.java

```diff
@@ -1343,4 +1346,67 @@ public void testDropNonExistingGenericTable() {
 
     genericTableApi.purge(currentCatalogName, namespace);
   }
+
+  @Test
+  public void testLoadTableWithSnapshots() {
+    Namespace namespace = Namespace.of("ns1");
+    restCatalog.createNamespace(namespace);
+    TableIdentifier tableIdentifier = TableIdentifier.of(namespace, "tbl1");
+    restCatalog.createTable(tableIdentifier, SCHEMA);
+
+    assertThat(catalogApi.loadTable(currentCatalogName, tableIdentifier, "ALL"))
+        .isEqualTo(OK.getStatusCode());
+    assertThat(catalogApi.loadTable(currentCatalogName, tableIdentifier, "all"))
+        .isEqualTo(OK.getStatusCode());
+    assertThat(catalogApi.loadTable(currentCatalogName, tableIdentifier, "refs"))
+        .isEqualTo(OK.getStatusCode());
+    assertThat(catalogApi.loadTable(currentCatalogName, tableIdentifier, "REFS"))
+        .isEqualTo(OK.getStatusCode());
+    assertThat(catalogApi.loadTable(currentCatalogName, tableIdentifier, "not-real"))
+        .isEqualTo(BAD_REQUEST.getStatusCode());
+
+    catalogApi.purge(currentCatalogName, namespace);
+  }
+
+  @Test
+  public void testLoadTableWithRefFiltering() {
+    Namespace namespace = Namespace.of("ns1");
+    restCatalog.createNamespace(namespace);
+    TableIdentifier tableIdentifier = TableIdentifier.of(namespace, "tbl1");
+
+    restCatalog.createTable(tableIdentifier, SCHEMA);
+
+    Table table = restCatalog.loadTable(tableIdentifier);
+
+    // Create an orphaned snapshot:
+    table.newAppend().appendFile(FILE_A).commit();
+    long snapshotIdA = table.currentSnapshot().snapshotId();
+    table.newAppend().appendFile(FILE_B).commit();
+    table.manageSnapshots().setCurrentSnapshot(snapshotIdA).commit();
+
+    String ns = RESTUtil.encodeNamespace(tableIdentifier.namespace());
+    try (Response res =
+        catalogApi
+            .request(
+                "v1/{cat}/namespaces/" + ns + "/tables/{table}",
+                Map.of("cat", currentCatalogName, "table", tableIdentifier.name()),
+                Map.of("snapshots", "all"))
+            .get()) {
+      LoadTableResponse responseContent = res.readEntity(LoadTableResponse.class);
+      assertThat(responseContent.tableMetadata().snapshots().size()).isEqualTo(2);
+    }
+
+    try (Response res =
+        catalogApi
+            .request(
+                "v1/{cat}/namespaces/" + ns + "/tables/{table}",
+                Map.of("cat", currentCatalogName, "table", tableIdentifier.name()),
+                Map.of("snapshots", "refs"))
+            .get()) {
+      LoadTableResponse responseContent = res.readEntity(LoadTableResponse.class);
+      assertThat(responseContent.tableMetadata().snapshots().size()).isEqualTo(1);
```

Comment:
can we also assert the intended snapshot id ? as if we got back the snapshot-id we wanted ?